src/models/beneficiary.js

```diff
@@ -84,19 +84,57 @@ class Beneficiary extends Model {
     };
   }
 
-  $afterValidate(json) {
-    super.$afterValidate(json);
-    const beneficiary = json;
-    const paymentList = beneficiary.paymentTypeList;
-
-    const result = paymentList.every((i) => paymentTypeEnum.includes(i));
+  static get relationMappings() {
+    return {
+      cases: {
+        relation: Model.HasManyRelation,
+        modelClass: `${__dirname}/Case`,
+        join: {
+          from: 'beneficiary.id',
+          to: 'case.beneficiaryId',
+        },
+      },
+      referees: {
+        relation: Model.ManyToManyRelation,
+        modelClass: `${__dirname}/Referee`,
+        join: {
+          from: 'beneficiary.id',
+          through: {
+            // persons_movies is the join table.
+            from: 'case.beneficiaryId',
+            to: 'case.refereeId',
+          },
+          to: 'referee.id',
+        },
+      },
+    };
+  }
 
-    console.log(result);
+  $afterValidate(beneficiary) {
+    super.$afterValidate(beneficiary);
+
+    // validate email
+    if (beneficiary.email !== undefined && beneficiary.email !== null) {
+      if (
+        /^[\w-.]+@([\w-]+\.)+[A-Za-z]{2,}$/.test(beneficiary.email) === false
+      ) {
+        throw new ValidationError({
+          message: `Email format "${beneficiary.email}" is invalid`,
+        });
```

Comment:
Do we have any common utils for these? Seen this in a couple of places now

Reply:
not sure, i adapted this from yitung's file for referees.